testsuite/views.py

```diff
@@ -1,37 +1,61 @@
 # pylint: disable=missing-docstring,unused-variable
-from flask import render_template, session
+from flask import jsonify, render_template, request, session
 from werkzeug import exceptions
 import flask_socketio
 
+from testsuite.application import app
+from testsuite.extensions import db, socketio
+from testsuite.models.testrun import TestRun
 from testsuite import tasks
 
 
-def configure_views(app, socketio):
+@app.route('/')
+def index():
+    return render_template('index.html')
 
-    @app.route('/')
-    def index():
-        return render_template('index.html')
 
-    @app.route('/authorized/')
-    def authorized():
-        exceptions.abort(500)
+@app.route('/load-report/<int:test_run_id>', methods=['POST'])
+def report(test_run_id):
+    room = request.form['room']
+    run = TestRun.query.get(test_run_id)
+    event = run.event
 
-    @socketio.on('connect')
-    def cb_handle_connect():
-        flask_socketio.send('connected')
+    socketio.emit('snapshot', event, room=room)
```

Comment:
Yeah, I was a little bit dreading figuring out how to re-render the report until I realized I could do this.